### ISSUE TITLE
Server crash after adding unknown users to permissions

### ISSUE THREAD

**Issue by shortybsd** [ForgeEssentials/ForgeEssentialsMain#1353]
After installing Forge Essentials, it takes 3 times to start the server where it will actually start successful. When it doesn't load and magically closes this is the error
````
[09:05:42] [Server thread/INFO] [FML/]: The state engine was in incorrect state ERRORED and forced into state SERVER_STOPPED. Errors may have been discarded.
[09:05:42] [Server thread/INFO] [FML/]: The state engine was in incorrect state ERRORED and forced into state AVAILABLE. Errors may have been discarded.
If it helps I am using screen via unix/linux running version 1.4.0 beta 5, forge version 1291 on java 7 64bit

Here is some more errors everytime it crashes the server upon start up
[09:11:00] [Server thread/ERROR] [FML/]: Fatal errors were detected during the transition from SERVER_ABOUT_TO_START to SERVER_STARTING. Loading cannot continue
[09:11:00] [Server thread/ERROR] [FML/]: 

different errors every time it starts;
[09:15:06] [Server thread/TRACE] [ForgeEssentials/ForgeEssentials]: Sending event FMLServerStartingEvent to mod ForgeEssentials
[09:15:07] [Server thread/ERROR] [FML/ForgeEssentials]: Exception caught during firing event com.forgeessentials.util.events.FEModuleEvent$FEModuleServerInitEvent@52b165b3:
java.lang.NullPointerException
````
I'm running version 1.4.0 beta 5, switching to 1.4.0 beta 6 and will let you know the outcome.

Update: beta 5 crashed and burned lol ... http://pastebin.com/Ld8NrXh3
downgrading to beta 3.
Update: beta 3 has same exact errors beta 6 stated.

**Comment by olee** [comment]
Probably duplicate of https://github.com/ForgeEssentials/ForgeEssentialsMain/issues/1333

**Comment by shortybsd** [comment]
What is crazy is it was just running. I changed a few things in the config restarted server now this.

**Comment by olee** [comment]
So does it now work?

**Comment by shortybsd** [comment]
Yes, all caused by adding users that haven't logged on the server. Please update the other guys of the fix and maybe change the if statement for the UID if it exists. Example you know Tommy is going to connect today and you want him to have permissions while you are gone. /p user Tommy group set members.

**Comment by luacs1998** [comment]
Can UserIdent be updated to also retrieve UUIDs from mojang servers given a name?

**Comment by olee** [comment]
I will try.
There was some stuff with UserIdent that I wanted to check anyway.

**Comment by olee** [comment]
Error log about this issue from IRC http://pastebin.com/raw.php?i=j9BsG2Jz

**Comment by shortybsd** [comment]
To replicate this exact issue, just add a user to a valid group that hasn't logged on the server. Use this for an example: p user Blahblah group set members. If Blahblah.txt didn't already exist under YourWorldDirectory\FEData\Permissions\Players (which will create automatically when the user connects to your server along with the valid uuid field) the UID field will not be added to that file when created via command therefore leading to a crash on your server's next restart. If you need examples on how it is achieved on adding users to the system and getting their username case-sensitive name as well as UID from login services have a look at PermissionsEx's source. I've used that plugin for years and added tons of people manually before they even connected to the server. When I get time I will go through your code to see if it automatically assumes that date value already exists in a player's file upon load, if so that is the problem and a simple fix.

**Comment by shortybsd** [comment]
Going through your source now I noticed you changed the SQL code, this error I am having is regarding the flatfile.

**Comment by shortybsd** [comment]
Found it - forgeessentials/permissions/persistence/FlatfileProvider.java Problem is the key fe.internal.player.known doesn't set they key or value. It should put something like this if the value isn't obtained:
fe.internal.player.known=false
Second the data field fe.internal.player.uuid is never created in the user file might be good to set it to a default value if it isn't obtained:
fe.internal.player.uuid=null then have the permissions on load to see if it can get a valid UI from login server. To help you even more I will create a "fake or non user that has not been on my server" and show you the output of the file comparing to a person that has logged in that I changed his/her group. http://pastebin.com/9H8t4tMn
